### PR TITLE
Bug fixing in find algorithm from ValueWithCountList

### PR DESCRIPTION
**Pull request by AndreiBozantan**

### CODE REVIEW

File: src/explorer/Common/ValueWithCountList.cs [hunk 1]
```diff
@@ -22,47 +22,51 @@ public void AddValueCount(T value, long count)
...
+                else if (count >= this[middle].Count)
                 {
-                    if (count <= this[middle + 1].Count)
+                    if (middle == right)
                     {
                         return this[middle].Value;
                     }
+                    if (count < this[middle + 1].Count)
+                    {
+                        return this[middle + 1].Value;
```
Comment (dandanlen) [inline src/explorer/Common/ValueWithCountList.cs:66]: Shouldn't this be:

```suggestion
                        return this[middle].Value;
```

(?)

Comment (AndreiBozantan) [reply src/explorer/Common/ValueWithCountList.cs:66]: I checked the code and the tests again and this looks good to me. Actually, the condition above this
```                    
if (middle == right)
{
    return this[middle].Value;
}
```
seems to be unnecessary.

Comment (dandanlen) [reply src/explorer/Common/ValueWithCountList.cs:66]: I don't really understand this. If `count < this[middle + 1].Count` is true, then surely this excludes the value at `[middle + 1]` since the `<` is non-inclusive? 

I would keep the `middle == right` condition - it seems to make sense, and I can think of scenarios where it might be triggered. Maybe the test cases are not exhaustive? I noticed the test cases are all evenly weighted - I think some of the edge cases would be triggered with non-even weighting. 

Let's merge this for now and we can try to write some better tests cases afterwards.
